src/main/java/uk/gov/digital/ho/hocs/queue/common/ComplaintService.java

```diff
@@ -66,25 +69,31 @@ public void createComplaint(ComplaintData complaintData, ComplaintTypeData compl
 
         caseworkClient.updateStageUser(caseUUID, stageForCaseUUID, UUID.fromString(clientContext.getUserId()));
 
-        caseworkClient.addCorrespondentToCase(caseUUID, stageForCaseUUID, complaintData.getComplaintCorrespondent());
+        ArrayList<ComplaintCorrespondent> correspondentsList = complaintData.getComplaintCorrespondent();
+        if (!correspondentsList.isEmpty()) {
 
-        UUID primaryCorrespondent = caseworkClient.getPrimaryCorrespondent(caseUUID);
+            for (ComplaintCorrespondent correspondent: correspondentsList) {
+                caseworkClient.addCorrespondentToCase(caseUUID, stageForCaseUUID, correspondent);
+            }
 
-        log.info("createComplaint, added primary correspondent : caseUUID : {}, primaryCorrespondent : {}", caseUUID, primaryCorrespondent);
+            UUID primaryCorrespondent = caseworkClient.getPrimaryCorrespondent(caseUUID);
 
-        Map<String, String> correspondents = Map.of(CORRESPONDENTS_LABEL, primaryCorrespondent.toString());
+            log.info("createComplaint, added primary correspondent : caseUUID : {}, primaryCorrespondent : {}", caseUUID, primaryCorrespondent);
 
-        workflowClient.advanceCase(caseUUID, stageForCaseUUID, correspondents);
+            Map<String, String> correspondents = Map.of(CORRESPONDENTS_LABEL, primaryCorrespondent.toString());
 
-        auditClient.audit(complaintTypeData.getCreateCorrespondentEventType(), caseUUID, stageForCaseUUID, correspondents);
+            workflowClient.advanceCase(caseUUID, stageForCaseUUID, correspondents);
 
-        log.info("createComplaint, case advanced for correspondent : caseUUID : {}", caseUUID);
+            auditClient.audit(complaintTypeData.getCreateCorrespondentEventType(), caseUUID, stageForCaseUUID, correspondents);
 
-        Map<String, String> complaintType = Map.of(COMPLAINT_TYPE_LABEL, complaintData.getComplaintType());
+            log.info("createComplaint, case advanced for correspondent : caseUUID : {}", caseUUID);
 
-        workflowClient.advanceCase(caseUUID, stageForCaseUUID, complaintType);
+            Map<String, String> complaintType = Map.of(COMPLAINT_TYPE_LABEL, complaintData.getComplaintType());
 
-        log.info("createComplaint, case advanced for complaintType : caseUUID : {}", caseUUID);
+            workflowClient.advanceCase(caseUUID, stageForCaseUUID, complaintType);
+
+            log.info("createComplaint, case advanced for complaintType : caseUUID : {}", caseUUID);
+        }
 
         log.info("createComplaint, completed : caseUUID : {}", caseUUID);
```

Comment:
Is there any additional logging required here for the instance where a case is created with no correspondents?

Reply:
I agree with Rob, a message would be good.